multiraft/transport.go

```diff
@@ -83,74 +92,63 @@ type localRPCTransport struct {
 // can be an arbitrary string). Each instance binds to a different unused port on
 // localhost.
 // Because this is just for local testing, it doesn't use TLS.
-func NewLocalRPCTransport() Transport {
+func NewLocalRPCTransport(stopper *stop.Stopper) Transport {
 	return &localRPCTransport{
-		listeners: make(map[proto.RaftNodeID]net.Listener),
-		clients:   make(map[proto.RaftNodeID]*rpc.Client),
-		conns:     make(map[net.Conn]struct{}),
-		closed:    make(chan struct{}),
+		servers: make(map[proto.RaftNodeID]*crpc.Server),
+		clients: make(map[proto.RaftNodeID]*netrpc.Client),
+		conns:   make(map[net.Conn]struct{}),
+		closed:  make(chan struct{}),
+		stopper: stopper,
 	}
 }
 
 func (lt *localRPCTransport) Listen(id proto.RaftNodeID, server ServerInterface) error {
-	rpcServer := rpc.NewServer()
-	err := rpcServer.RegisterName("MultiRaft", server)
-	if err != nil {
-		return err
-	}
-
-	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	addr := util.CreateTestAddr("tcp")
+	rpcServer := crpc.NewServer(addr, &crpc.Context{
+		Context: base.Context{
+			Insecure: true,
+		},
+		Stopper:      lt.stopper,
+		DisableCache: true,
+	})
+	err := rpcServer.RegisterAsync(raftMessageName,
```

Comment:
I'd like us to embrace this everywhere; the explicitness of intent outweighs aesthetics in my opinion.

The above is just a rant, merge when you're ready and thanks for fixing this.